code/datums/components/crafting/crafting.dm

```diff
@@ -323,14 +345,19 @@
 	while(Deletion.len)
 		var/DL = Deletion[Deletion.len]
 		Deletion.Cut(Deletion.len)
-		// Snowflake handling of reagent containers and storage atoms.
+		// Snowflake handling of reagent containers, storage atoms, and structures with contents.
 		// If we consumed them in our crafting, we should dump their contents out before qdeling them.
 		if(is_reagent_container(DL))
 			var/obj/item/reagent_containers/container = DL
 			container.reagents.expose(container.loc, TOUCH)
 		else if(istype(DL, /obj/item/storage))
 			var/obj/item/storage/container = DL
 			container.emptyStorage()
+		else if(isstructure(DL))
+			var/obj/structure/structure = DL
+			if(structure.contents.len)
+				for(var/obj/item in structure.contents)
+					item.forceMove(item.drop_location())
```

Comment:
This makes me a bit nervous, there might be stuff in the contents never meant to see the light of day. Could you call dump_contents instead?

Reply:
> This makes me a bit nervous, there might be stuff in the contents never meant to see the light of day. Could you call dump_contents instead?

Fair, done!